tests/unit/backend/chalice/api_server/test_revisions.py

```diff
@@ -116,3 +120,202 @@ def test__publish_revision__201(self):
             self.assertIn(link, actual_body["links"])
         for keys in expected_body.keys():
             self.assertEqual(expected_body[keys], actual_body[keys])
+
+    def test__revision_deleted_with_published_datasets(self):
+        """The published dataset artifacts should be intact after deleting a collection revision"""
+        # Generate public collection
+        pub_collection = self.generate_collection(self.session, visibility="PUBLIC")
+        pub_dataset = self.generate_dataset(
+            self.session, collection_visibility="PUBLIC", collection_id=pub_collection.id, published=True
+        )
+        dep_dir = self.generate_deployment_directory(self.session, pub_dataset.id, upload=True)
+        arts = [
+            self.generate_artifact(self.session, pub_dataset.id, ext, upload=True) for ext in DatasetArtifactFileType
+        ]
+        expected_body = json.loads(json.dumps(pub_collection.reshape_for_api(), cls=CustomJSONEncoder))
+
+        # Generate revision
```

Comment:
Why are you doing this manually instead of using collection.revision function?

Reply:
Somehow I forget we had that